front/app/components/admin/AddCollaboratorsModal/AddCollaboratorsModal.test.tsx

```diff
@@ -51,53 +51,134 @@ jest.mock('api/seats/useSeats', () => () => {
+        screen.queryByText(
+          'You have reached the limit of included seats within your plan, 1 additional seat will be added.'
+        )
+      ).toBeInTheDocument();
+      expect(buyAdditionalSeatButton).toBeInTheDocument();
+    });
   });
 
-  it('shows buy an additional seat in button when seats are full and admin is trying to add another', () => {
-    mockUserSeatsData.data.attributes.project_moderators_number = 10;
+  describe('when seat_based_billing is on', () => {
+    beforeEach(() => {
+      mockFeatureFlagData = true;
+    });
 
-    render(
-      <AddCollaboratorsModal
-        addModerators={handleOnAddModeratorsClick}
-        showModal
-        closeModal={closeModal}
-      />
-    );
+    it('shows confirm in button when seats are more than the ones set in limit but within the sum of additional seats and maximum seats', () => {
+      mockUserSeatsData.data.attributes.project_moderators_number = 10;
+      mockAppConfiguration.data.attributes.settings.core.additional_moderators_number = 7;
+      mockFeatureFlagData = true;
+      render(
+        <AddCollaboratorsModal
+          addModerators={handleOnAddModeratorsClick}
+          showModal
+          closeModal={closeModal}
+        />
+      );
 
-    const buyAdditionalSeatButton = screen.queryByRole('button', {
-      name: 'Buy 1 additional seat',
+      const confirmButton = screen.queryByRole('button', {
+        name: 'Confirm',
+      });
+
+      expect(
+        screen.queryByText('Give collaborator rights')
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByText(
+          'Are you sure you want to give 1 person collaborator rights?'
+        )
+      ).toBeInTheDocument();
+      expect(confirmButton).toBeInTheDocument();
     });
 
-    expect(screen.queryByText('Give collaborator rights')).toBeInTheDocument();
-    expect(
-      screen.queryByText(
-        'You have reached the limit of included seats within your plan, 1 additional seat will be added.'
-      )
-    ).toBeInTheDocument();
-    expect(buyAdditionalSeatButton).toBeInTheDocument();
+    it('shows buy an additional seat in button when seats are more than the sum of additional seats and maximum seats', () => {
+      mockUserSeatsData.data.attributes.project_moderators_number = 12;
+      mockAppConfiguration.data.attributes.settings.core.additional_moderators_number = 1;
+      mockFeatureFlagData = true;
+      render(
+        <AddCollaboratorsModal
+          addModerators={handleOnAddModeratorsClick}
+          showModal
+          closeModal={closeModal}
+        />
+      );
+
+      const buyAdditionalSeatButton = screen.queryByRole('button', {
+        name: 'Buy 1 additional seat',
+      });
+
+      expect(
+        screen.queryByText('Give collaborator rights')
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByText(
+          'You have reached the limit of included seats within your plan, 1 additional seat will be added.'
```

Comment:
In general, wondering if we should tie the test this closely to copy. For each change, we have to change the tests.